### DISCUSSION TITLE
The remote server does not speak TLS - please help!

### DISCUSSION THREAD

**Discussion by yuval1973** [mitmproxy/mitmproxy#6755] [General]
Hi,

I'm trying to use mitmproxy to intercept the traffic before an external (upstream) proxy as follows:

`mitmproxy --mode upstream:http://the-external-proxy:12345 --upstream-auth theUsername:thePassword`

but then **every** request that is done via mitmproxy to **any** website ends with: **The remote server does not speak TLS error**. for example:

`curl -x http://localhost:8080 -k https://anysite.com` 

gives:

`502 Bad Gateway - The remote server does not speak TLS`

This should be straightforward but yet it fails and I spent hours to understand why to no avail. Surely something I'm doing wrong here or forgot to set as this should work right out of the bat from a simple command line. (and yes I already tried to set the connection strategy to lazy but this doesn't fix the problem)

using `Version: 10.2.4`

**Comment by rosydawn6** [comment]
You could try doing a standard setup without any upstream proxy to make sure all other aspects of the infrastructure/topology are correct.

**Comment by yuval1973** [comment]
this was turned to be a problem with the particular remote proxy. closing the issue

**Reply by unpacify** [reply]
did you ever find a proxy that doesnt have this issue with the upstream??? im using IPRoyal and im having the same issue

**Comment by augustdua** [comment]
Having the exact same issue while using uTLS above mitmprxoy
